File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program99.cpp
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and check whether it contains vowels in it or not.				
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{

	private:
		string str;
		int iCnt,iCount;
		bool bAns;

	public:

		Solution(string str,int iCnt=0,int iCount=0,bool bAns=false)
		{
			this->str=str;
		}

		bool ChkVowel()
		{
			while(iCnt!=str.length())
			{
				if(str[iCnt]=='a' or str[iCnt]=='e' or str[iCnt]=='i' or str[iCnt]=='o' or str[iCnt]=='u')
				{
					bAns = true;
					break;
				}

				iCnt++;
			
			}	
			return bAns;
		}
};

int main()
{
	string str;
	bool bRet=false;

	cout<<"Enter string: ";
	getline(cin,str);
	
	Solution *sObj = new Solution(str);
	bRet = sObj->ChkVowel();
	if(bRet==true)
	{
		cout<<"Vowel is present\n";
	}
	else
	{
		cout<<"Vowel is absent\n";
	}

	delete sObj;
	return 0;

}

/*
OUTPUT:

Enter string: marvellous
Vowel is present

Enter string: xyz
Vowel is absent

*/

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program99.java
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and check whether it contains vowels in it or not.				
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Solution
{
	String str;
	int iCnt;
	boolean bAns;

	Solution(String str)
	{
		this.str=str;
		this.iCnt=0;
		this.bAns = false;
	}

	boolean ChkVowel()
	{
		
		while(iCnt!=str.length())
		{
			if(str.charAt(iCnt)=='a' || str.charAt(iCnt)=='e' || str.charAt(iCnt)=='i' || str.charAt(iCnt)=='o' || str.charAt(iCnt)=='u')
			{
				bAns = true;
				break;
			}
			iCnt++;
		}
		
		return bAns;
	}
}

class Program
{
	public static void main(String args[])
	{
		boolean bRet=false;
		Scanner sObj = new Scanner(System.in);
		System.out.println("Enter the string: ");
		String str = sObj.nextLine();

		Solution obj = new Solution(str);
		bRet = obj.ChkVowel();

		if(bRet==true)
		{
			System.out.println("Vowel is present");
		}
		else
		{
			System.out.println("Vowel is absent");
		}
	}
}

/*
OUTPUT:
Enter the string: 
Vishal
Vowel is present

Enter the string: 
xyz
Vowel is absent

*/
